internal/monitors/docker/docker.go

```diff
@@ -225,9 +234,63 @@ func parseContainerEnvSlice(env []string) map[string]string {
 	return out
 }
 
+// EnableExtraGroups enables extra metrics that were individually turned on
+// by ExtraMetrics/ExtraGroups configuration
+func EnableExtraGroups(initConf EnhancedMetricsConfig, extraGroups, extraMetrics []string) EnhancedMetricsConfig {
+	groupEnableMap := map[string]bool{
+		groupBlkio: initConf.EnableExtraBlockIOMetrics,
+		groupCPU: initConf.EnableExtraCPUMetrics,
+		groupMemory: initConf.EnableExtraMemoryMetrics,
+		groupNetwork: initConf.EnableExtraNetworkMetrics,
+	}
+
+	for _, group := range extraGroups {
+		groupEnableMap[group] = true
+	}
+	for _, metric := range extraMetrics {
+		if metricSet[metric].Group != "" {
+			groupEnableMap[metricSet[metric].Group] = true
```

Comment:
We need to do the thing [I referenced above](https://github.com/signalfx/signalfx-agent/pull/738/files#r278372837) so that it handles globs and (and excluded metrics for free). @keitwb thoughts on the implementation details of it?